Chore: appease linter

BlaineHeffron opened the pull request:
A number of fixes to remove the warnings and errors across the src/ directory of the project.

I noticed the `src/.eslintrc.js` files was not actually being used when running `yarn fmt`. Since `yarn fmt` only runs the linter in the `src` directory anyway, I just merged the  `src/.eslintrc.js` file with the top level `.eslintrc.js` file.

Then, I added the settings below to be ignored. These were mostly minor docstring formatting complaints or some other trivial things like having spaces in comments. 

```
 "jsdoc/require-param-type": 0,
 "jsdoc/require-returns-type": 0,
 "jsdoc/no-blank-blocks": 0, 
 "jsdoc/no-multi-asterisks": 0,
 "jsdoc/tag-lines": "off",
 "jsdoc/require-jsdoc": "off",
 "valid-jsdoc": "off",
 "import/extensions": 0,
"spaced-comment": 0,
```

I then fixed all other issues. The only logic changes are where functions contained for loops, I used iterators instead (e.g forEach, some, filter, etc). There was one case where I added eslint-ignore comments for a double nested for loop that was a bit more complicated so I just decided to leave it be. 

Sometimes there were names prefixed with an underscore, in those cases I just removed the underscore.

chadoh commented on config/tsconfig.json:
```diff
@@ -11,5 +11,8 @@
-  "include": ["../src"]
+  "include": [
+    "../src",
+    "../src/.eslintrc.js"
```
This should be removed now, right?

BlaineHeffron replied:
yes

chadoh reviewed (approved):
Amazing!!! Thanks so much!

Shaptic commented:
@BlaineHeffron I'm too scared to touch the test conflict :sweat_smile:

BlaineHeffron commented on src/contract/spec.ts:
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable no-fallthrough */
+/* eslint-disable default-case */
```
I changed it in this file to target each issue individually

BlaineHeffron commented:
Addressed @chadoh comments, performed rebase, and then added yarn fmt fixes after the rebase. The main fix applied after the rebase was a dependency cycle involving DEFAULT_TIMEOUT and NULL_ACCOUNT. I moved these to types.ts to fix the issue.

chadoh reviewed (approved):
Looks great, let's get it in fast so we don't need to do it again!

Shaptic commented on src/rpc/server.ts:
```diff
@@ -153,20 +190,19 @@ export class Server {
    *    {@link Address} instance
    * @param {xdr.ScVal} key   the key of the contract data to load
-   * @param {Durability} [durability=Durability.Persistent]   the "durability
+   * @param {Durability} [durability]   the "durability
```
Shouldn't we keep the default?

BlaineHeffron replied:
Seems it was auto removing the default due to jsdoc/no-defaults. I turned that off and restored the defaults in the docs.

Shaptic commented on src/rpc/server.ts:
```diff
@@ -153,20 +190,19 @@ export class Server {
    *
    * @returns {Promise<Api.LedgerEntryResult>}   the current data value
    *
-   * @warning If the data entry in question is a 'temporary' entry, it's
```
TIL that the warning block was [never getting rendered](https://stellar.github.io/js-stellar-sdk/Server.html#getContractData) :facepalm:

Shaptic commented on src/webauth/utils.ts:
```diff
@@ -29,7 +32,7 @@ import { ServerApi } from "../horizon/server_api";
  * @param {string} homeDomain The fully qualified domain name of the service
  *    requiring authentication
- * @param {number} [timeout=300] Challenge duration (default to 5 minutes).
+ * @param {number} [timeout] Challenge duration (default to 5 minutes).
```
same thing here: we should keep the default?

Shaptic commented on src/webauth/utils.ts:
```diff
@@ -133,6 +137,97 @@ export function buildChallengeTx(
+ * transaction.sign(keypair1, keypair2)
+ * WebAuth.gatherTxSigners(transaction, [keypair1.publicKey(), keypair2.publicKey()])
+ */
+export function gatherTxSigners(
```
Did anything change in these? Since they're so security-sensitive I'm praying there are no bugs introduced by some sort of eslint-compatibility changes

BlaineHeffron replied:
Mainly just reordering function calls so that the definitions come before they are called. Some for loops were changed, I'll make a separate PR for these.

Shaptic commented on src/webauth/utils.ts:
```diff
@@ -633,19 +595,19 @@ export function verifyChallengeTxSigners(
   let serverSignatureFound = false;
   let clientSigningKeySignatureFound = false;
-  for (const signer of signersFound) {
+  signersFound.forEach((signer) => {
```
Why is this preferred when it's technically slower due to add'l function call stacks?

BlaineHeffron replied:
I'm not sure because you are correct about it being slower. The rule description is `   iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations  no-restricted-syntax`. In general I tried to change the for loops to something that made sense like filter but in some cases you do need to iterate over all which is why I just subbed in forEach. I could do an eslint ignore for those cases. For now I'll separate the webauth utils change into a separate PR.

Shaptic replied:
Oh wow interesting, good to know!

Shaptic reviewed (commented):
There's so much going on here :sob: but I think I covered all of the bases. I'd feel more comfortable with keeping the `WebAuth` stuff unchanged for now for security reasons and reviewing that more carefully in a separate PR that aims to minimize the diff so that we can appropriately evaluate the changes. Could you reset that?

BlaineHeffron commented on src/horizon/account_call_builder.ts:
```diff
@@ -8,8 +8,8 @@ import { ServerApi } from "./server_api";
- * @extends CallBuilder
- * @constructor
+ * @augments CallBuilder
+ * @class
```
Yes these are auto fixes
@augments and @extends are synonyms, so I guess its auto fixing to the newer one. https://jsdoc.app/tags-augments
Seems the same applies to @class / @constructor - they are synonyms https://jsdoc.app/tags-class

BlaineHeffron commented on src/horizon/account_response.ts:
```diff
@@ -9,47 +10,69 @@ import { ServerApi } from "./server_api";
   public readonly last_modified_ledger!: number;
+
   public readonly last_modified_time!: string;
+
```
Sure I'll ignore whatever the rule is that is doing this... I had already removed a bunch of these rules but I guess I missed some.

BlaineHeffron replied:
Fixed

BlaineHeffron commented on src/horizon/call_builder.ts:
```diff
@@ -195,7 +204,7 @@ export class CallBuilder<
    * Sets `limit` parameter for the current call. Returns the CallBuilder object on which this method has been called.
    * @see [Paging](https://developers.stellar.org/api/introduction/pagination/)
-   * @param {number} number Number of records the server should return.
+   * @param "recordsNumber" number Number of records the server should return.
```
This was a typo on my end, its now fixed.

BlaineHeffron commented:
Reverted webauth/utils.ts and addressed the other comments. Should be good to go.

chadoh commented:
I think you need to merge, @Shaptic!
